sklearn/preprocessing/_function_transformer.py

```diff
@@ -245,22 +250,8 @@ def transform(self, X):
             # check the consistency between the column names of the output and the
             # one generated by `get_feature_names_out`
             if list(out.columns) != list(self.get_feature_names_out()):
-                raise ValueError(
-                    "The output generated by `func` have different column names than "
-                    "the one generated by the method `get_feature_names_out`. "
-                    f"Got output with columns names: {list(out.columns)} and "
-                    "`get_feature_names_out` returned: "
-                    f"{list(self.get_feature_names_out())}. "
-                    "This can be fixed in different manners depending on your use case:"
-                    "\n(i) If `func` returns a container with column names, make sure "
-                    "they are consistent with the output of `get_feature_names_out`.\n"
-                    "(ii) If `func` is a NumPy `ufunc`, then forcing `validate=True` "
-                    "could be considered to internally convert the input container to "
-                    "a NumPy array before calling the `ufunc`.\n"
-                    "(iii) The column names can be overriden by setting "
-                    "`set_output(transform='pandas')` such that the column names are "
-                    "set to the names provided by `get_feature_names_out`."
-                )
+                # TODO: should we trigger a copy or not?
+                out.columns = self.get_feature_names_out()
```

Comment:
I think we better the container (pandas or polars or whatever), and then construct a new one with the columns, rather than triggering a `copy`.

Reply:
> I think we better the container (pandas or polars or whatever)

I thin that I am missing a verb here :)

Reply:
updated 😁